main.go

```diff
@@ -49,6 +49,23 @@ var Options struct {
 	// this will be used for API calls back to the Assisted Service API
 	// Will default to the value held in HTTPS_CA_FILE unless overridden
 	AssistedServiceApiTrustedCAFile string `envconfig:"ASSISTED_SERVICE_API_TRUSTED_CA_FILE" default:""`
+
+	// OSImageDownloadHeaders contains a JSON encoded representation of any
+	// HTTP headers to be sent with every request to download an OS image.
+	OSImageDownloadHeaders string `envconfig:"OS_IMAGES_HTTP_REQUEST_HEADERS" default:""`
+	// OSImageDownloadQueryParams contains a JSON encoded representation of any
+	// query parameters to be sent with every request to download an OS image.
+	OSImageDownloadQueryParams string `envconfig:"OS_IMAGES_HTTP_REQUEST_QUERY_PARAMS" default:""`
+}
+
+func unmarshallJSONMap(jsonMap string) (map[string]string, error) {
```

Comment:
I thought envconfig handled maps? Did you look into that?

Reply:
Ah, answered my own question. It doesn't quite use json  https://github.com/kelseyhightower/envconfig/blob/v1.4.0/envconfig.go#L319-L342

Probably best to stick with json then.
Also I suppose we're doing this already for OS Images :shrug:

Reply:
Yeah, just changed it now to get rid of this